generated/niscope/session.py

```diff
@@ -2302,6 +2302,21 @@ def fetch(self, num_samples, timeout=5.0):
         '''
         return self._fetch(num_samples, timeout)
 
+    def fetch_array(self, num_samples, timeout=5.0):
+        import array
+
+        vi_ctype = visatype.ViSession(self._vi)  # case 1
+        channel_list_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case 2
+        timeout_ctype = visatype.ViReal64(timeout)  # case 9
+        num_samples_ctype = visatype.ViInt32(num_samples)  # case 9
+        wfm_array = array.array("d", range(num_samples * self._actual_num_wfms()))
```

Comment:
This initializes the array as [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0,... which surely isn't what you intended and may explain the crappy performance.

So... one way to initialize to zeroes that may be marginally faster than actually using the iterator returned by range which returns an int that also needs to be cast is to write our own (untested):

'''
class Zeroes:
    def __init__(self):
        pass
    	
    def __iter__(self):
        return self

    def next(self): # Python 3: def __next__(self)
        return 0.0
 '''

I'm looking for a cheaper way, but I'm guessing you looked as well?

Reply:
I'm guessing that a combination of bytearray, memoryview, etc. might do the trick. Need to play with it though.

Reply:
Changed to `wfm_array = array.array("d", [0]) * (num_samples * self._actual_num_wfms())`